### PR TITLE
PLNSRVCE-420 : Implement Tekton results for openshift development

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
Implement Tekton Results and Postgresql DB for openshift development

Signed-off-by: Cathal O'Connor [cathal@redhat.com](mailto:cathal@redhat.com)

### CODE REVIEW

File: ckcp/tekton-results/openshift/configmap.yaml
Comment (Roming22) [inline ckcp/tekton-results/openshift/configmap.yaml:1]: All yaml must start with `---`, otherwise yamllint will complain in the CI. Occurs multiple times in the PR.

File: ckcp/tekton-results/openshift/kustomization.yaml
Comment (Roming22) [inline ckcp/tekton-results/openshift/kustomization.yaml:1]: Please move `ckcp/tekton-results` to `ckcp/manifests/tekton-results`

File: images/access-setup/content/bin/setup_compute.sh
Comment (Roming22) [inline images/access-setup/content/bin/setup_compute.sh:187]: Add `-s` for secret. You probably need `-r` as well so that shellcheck does not complain.

File: ckcp/openshift_dev_setup.sh
Comment (Roming22) [inline ckcp/openshift_dev_setup.sh:389]: I would expect `"$WORK_DIR/manifests/compute/tekton-results/tekton-results-secret.yaml"` to be created for the user at that point.
It would then be applied by `setup_compute.sh`.

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh]: Commented for testing, I've removed commented code now as it was blocking the request for username/password from displaying

File: ckcp/tekton-results/openshift/configmap.yaml
Comment (CathalOConnorRH) [reply ckcp/tekton-results/openshift/configmap.yaml]: Updated yaml

File: ckcp/tekton-results/openshift/kustomization.yaml
Comment (CathalOConnorRH) [reply ckcp/tekton-results/openshift/kustomization.yaml]: Moved.

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh]: added both options

File: ckcp/openshift_dev_setup.sh
Comment (CathalOConnorRH) [reply ckcp/openshift_dev_setup.sh]: removed create secret as it shouldn't be needed.

File: images/access-setup/content/bin/setup_compute.sh
Comment (Roming22) [inline images/access-setup/content/bin/setup_compute.sh:184]: This var seems to be unused and could be removed (L184,L185,L194).

File: images/access-setup/content/bin/setup_compute.sh
Comment (Roming22) [inline images/access-setup/content/bin/setup_compute.sh:182]: I would expect that manifest to be applied in https://github.com/openshift-pipelines/pipeline-service/blob/main/images/cluster-setup/bin/install.sh#L133

Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh:182]: added now.

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh]: var removed

**Comment by Roming22** [comment]
@bnallapeta @Mo3m3n @ramessesii2 The PR is quite large, so I'd like another pair of eyes to go through it before it's merged.

**Comment by CathalOConnorRH** [comment]
@Mo3m3n  I've added a README to the tekton-results section in the gitops directory.

File: ckcp/openshift_dev_setup.sh
Comment (CathalOConnorRH) [reply ckcp/openshift_dev_setup.sh:92]: This PR covers two different install modes where the requirements are different. In a ckcp install we need to configure a postgres db instance where as on workload clusters we will be using an external database.

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh:146]: This had to be removed so that the user could be asked for credentials for configuring the database. If this is left in, the script hangs with no output until the credentials are entered and then the output is displayed which would be too late.

File: ckcp/openshift_dev_setup.sh
Comment (Mo3m3n) [reply ckcp/openshift_dev_setup.sh]: Understood, so you're right this part is only relevant to local dev environment.
In such case, what about moving this task to `install_pipeline_service()`. This is where we handle deploying pipeline service in a local ckcp cluster and includes all the necessary bricks. IMO since the only purpose of this db is to be used with tekton_results then we can apply its manifets in `install_pipeline_service()`

Comment (CathalOConnorRH) [reply ckcp/openshift_dev_setup.sh]: sure, updated now.

File: images/access-setup/content/bin/setup_compute.sh
Comment (Mo3m3n) [reply images/access-setup/content/bin/setup_compute.sh:146]: Ok I would have preferred to keep the possibility to run ckcp script in a non interactive way so we can just run it and forget about it. But we can handle this in a later task.

Comment (Roming22) [reply images/access-setup/content/bin/setup_compute.sh:146]: There should be no user interaction when using the script (both for UX, and because it's used by the CI bot). Please find a way to inject the credentials.

Comment (Mo3m3n) [reply images/access-setup/content/bin/setup_compute.sh:146]: Actually I am wondering if this is going to work with github test
@Roming22 @xinredhat if you confirm that github-ci is running a fresh ckcp cluster then the script is going to hang for user input.

**Comment by Roming22** [comment]
@CathalOConnorRH Please update `ckcp/test.sh` to add a `results` test that retrieve something from the DB (can be based on the pipelines test). There's currently no way to validate that the service was setup properly.

File: gitops/argocd/tekton-results/README.md
Comment (avinal) [inline gitops/argocd/tekton-results/README.md:6]: Should we rather point to our fork?

File: images/access-setup/content/bin/setup_compute.sh
Comment (avinal) [reply images/access-setup/content/bin/setup_compute.sh:146]: I don't know if this makes sense, but can we provide credentials using config.yaml?

File: images/access-setup/content/bin/setup_compute.sh
Comment (avinal) [reply images/access-setup/content/bin/setup_compute.sh:187]: yes this is possible, even in the installation instructions we have a db with random password: 

https://github.com/openshift-pipelines/tektoncd-results/blob/172d1b55bf7cfe62bfe1c7a0071cf9820a2dadb1/docs/install.md?plain=1#L24

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh]: I've updated to now pass tekton details as arguments or generate if not passed into the scripts

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh]: updated now

File: gitops/argocd/tekton-results/README.md
Comment (CathalOConnorRH) [reply gitops/argocd/tekton-results/README.md:6]: I think it'll be cleaner to confirm this PR works as expected and then update what's needed for our fork and the included changes we may need once this is merged.

File: ckcp/openshift_dev_setup.sh
Comment (Roming22) [inline ckcp/openshift_dev_setup.sh:351]: With the latest changes, my understanding is that this change can be reverted.

File: ckcp/openshift_dev_setup.sh
Comment (Roming22) [inline ckcp/openshift_dev_setup.sh:385]: Do not redirect the output, instead use `indent` to format it properly.

File: ckcp/test.sh
Comment (Roming22) [inline ckcp/test.sh:217]: That seems like a simple `a=b`. Do you expect `RECORD_CMD` to be executed?

File: ckcp/test.sh
Comment (Roming22) [inline ckcp/test.sh:220]: Shellcheck is likely to report an error about missing '"' here. Please double check, not sure I understand why an error was not reported.

File: ckcp/openshift_dev_setup.sh
Comment (CathalOConnorRH) [reply ckcp/openshift_dev_setup.sh]: Reverted now.

File: ckcp/openshift_dev_setup.sh
Comment (CathalOConnorRH) [reply ckcp/openshift_dev_setup.sh]: replaced with Indent

File: ckcp/test.sh
Comment (CathalOConnorRH) [reply ckcp/test.sh]: Good spot. I've updated now and confirmed the record is returned correctly

File: ckcp/test.sh
Comment (CathalOConnorRH) [reply ckcp/test.sh]: There is no error with or without the quote, I think shellcheck can allow some commands without quotes. I've added them for completeness and to ensure no future issues if they are detected at a later stage.

File: images/access-setup/content/bin/setup_compute.sh
Comment (Roming22) [inline images/access-setup/content/bin/setup_compute.sh:77]: I'm not sure those params should have default values.

File: images/access-setup/content/bin/setup_compute.sh
Comment (Roming22) [inline images/access-setup/content/bin/setup_compute.sh:205]: The TEKTON_RESULTS_* vars should probably be checked before generating the secret, and an error returned if unset.

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh]: defaults removed

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh:210]: added check and error output

**Comment by adambkaplan** [comment]
Current failure:

```
ekton-chains signing-secrets --from-file=/tmp/tmp.YXhWwhhvek/credentials/manifests/compute/tekton-chains/tmp --dry-run=client -o yaml
        + yq '. += {"immutable" :true}'
        + yq 'sort_keys(.)'
        + rm -rf /tmp/tmp.YXhWwhhvek/credentials/manifests/compute/tekton-chains/tmp
        ++ yq .data
        ++ grep -cE '^cosign.key:|^cosign.password:|^cosign.pub:'
        + '[' 3 '!=' 3 ']'
        + printf 'OK\n'
        OK
    + printf -- '  - tekton-results manifest:\n'
      - tekton-results manifest:
    + tekton_results_manifest
    + indent 4
    ++ printf %4s
    + sed 's/^/    /'
        + manifest=/tmp/tmp.YXhWwhhvek/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml
        + '[' '!' -e /tmp/tmp.YXhWwhhvek/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml ']'
        + [[ -z tekton ]]
        + [[ -z i5pjaEMiv5fXAfmzay6mnI7Oorg= ]]
        //images/access-setup/content/bin/setup_compute.sh: line 200: /tmp/tmp.YXhWwhhvek/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml: No such file or directory
        ++ yq .data
        //images/access-setup/content/bin/setup_compute.sh: line 216: /tmp/tmp.YXhWwhhvek/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml: No such file or directory
        ++ grep -cE '^DATABASE_USER:|^DATABASE_PASSWORD:'
        + '[' 0 '!=' 2 ']'
        + printf '[ERROR] Invalid manifest: '\''%s'\''' /tmp/tmp.YXhWwhhvek/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml
        [ERROR] Invalid manifest: '/tmp/tmp.YXhWwhhvek/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml'+ exit 1
command terminated with exit code 1
```

**Comment by Roming22** [comment]
@CathalOConnorRH You need to change the regex to "^ *DATABASE_" as the value is indented and we don't care by how much.

File: ckcp/openshift_dev_setup.sh
Comment (Mo3m3n) [inline ckcp/openshift_dev_setup.sh:37]: @CathalOConnorRH Can you please move the database params in `config/config.yaml`.
In "openshift_dev_setup.sh" we only deal with global params, but for specific app configs we centralise that in "config/config.yaml" otherwise we may end up with a myriad of script args for the different apps to configure.
Smth like the following would work:
```yaml
tekton_results_db:
  user:
  password:
```
You can find in the ckcp script how we parse that file.

**Comment by Mo3m3n** [comment]
> /lgtm @Mo3m3n Please check if your comments have been answered. If you don't see any other issues, please merge.

We're almost good, one last review I have submitted then will try this locally and merge it if the test is successful

File: ckcp/openshift_dev_setup.sh
Comment (Mo3m3n) [inline ckcp/openshift_dev_setup.sh:390]: You will need to prefix `$CKCP_DIR` otherwise this is not going to work if run outside of ckcp folder

File: images/access-setup/content/bin/setup_compute.sh
Comment (Mo3m3n) [inline images/access-setup/content/bin/setup_compute.sh:218]: no need for wildcards (they are producing warnings) just use `grep -cE "DATABASE_USER|DATABASE_PASSWORD"` it should work fine

**Review by Mo3m3n — CHANGES_REQUESTED** [review]
currently commit message is `Merge branch 'main' into PLNSRVCE-420`
You will need to change it to relate to what the PR is about

File: images/access-setup/content/bin/setup_compute.sh
Comment (CathalOConnorRH) [reply images/access-setup/content/bin/setup_compute.sh]: updated now.

File: ckcp/openshift_dev_setup.sh
Comment (CathalOConnorRH) [reply ckcp/openshift_dev_setup.sh]: updated now

**Review by Mo3m3n — APPROVED** [review]

**Pull request merged**
